qiskit_experiments/calibration_management/calibration_utils.py

```diff
@@ -12,51 +12,137 @@
 
 """Calibration helper functions"""
 
-from typing import List, Set
-from qiskit.pulse import ScheduleBlock, Call
+from typing import Optional, Set, Tuple
+import regex as re
+import retworkx as rx
 
+from qiskit.circuit import ParameterExpression, Parameter
+from qiskit.pulse import ScheduleBlock
 
-def used_in_calls(schedule_name: str, schedules: List[ScheduleBlock]) -> Set[str]:
-    """Find the schedules in the given list that call a given schedule by name.
+from qiskit_experiments.exceptions import CalibrationError
+
+
+def update_schedule_dependency(schedule: ScheduleBlock, dag: rx.PyDiGraph):
```

Comment:
I would leave it here. You will have issues in Terra with this because you will have to deal with both assigned and unassigned references. The calibrations in QE forces unassigned references which makes this possible.